crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -3134,25 +3156,34 @@ fn escape_with_source(
     s: &str,
     single_quote: Option<bool>,
 ) -> String {
-    if target <= EsVersion::Es5 {
-        return escape_without_source(s, target, single_quote.unwrap_or(false));
-    }
+    static UNICODE_CODEPOINT: Lazy<regex::Regex> =
+        Lazy::new(|| regex::Regex::new(r#"(?i)\\[u]"#).unwrap());
 
     if span.is_dummy() {
-        return escape_without_source(s, target, single_quote.unwrap_or(false));
+        return escape_without_source(s, target, single_quote.unwrap_or(false), false);
     }
 
-    //
     let orig = cm.span_to_snippet(span);
     let orig = match orig {
         Ok(orig) => orig,
         Err(v) => {
-            return escape_without_source(s, target, single_quote.unwrap_or(false));
+            return escape_without_source(s, target, single_quote.unwrap_or(false), false);
         }
     };
 
+    if target <= EsVersion::Es5 {
+        let emit_non_ascii_as_unicode = UNICODE_CODEPOINT.is_match(&orig);
```

Comment:
The only concern about this PR I have is the performance of this line.
Do we really need regex?

Reply:
@kdy1 I think that's my JS brain defaulting to regex 😛  Simple string contains is enough, updated in [4a11bab](https://github.com/swc-project/swc/pull/3636/commits/4a11bab72e265ab7cdeecb1746262a6c48b317fb)